content/blog/2022-05-04-e2e-cv-pt2.md

```diff
@@ -0,0 +1,180 @@
+Stage 'data_load' didn't change, skipping
+Stage 'data_split' didn't change, skipping
+Stage 'train' didn't change, skipping
+Running stage 'evaluate':
+> python src/stages/eval.py --config=params.yaml
+...
+```
+
+There is a super convenient set of
+[Experiment Management](https://dvc.org/doc/user-guide/experiment-management)
+features that makes switching between reproducible experiments very easy without
+adding failed experiments to your git history. Check out this
+[blog post](https://dvc.org/blog/ml-experiment-versioning) which talks about the
+idea of "ML Experiments as Code". That means treating experiments as you’d treat
+code i.e. use git to track all changes in configs, metrics, and data versions
+through text files. This approach removes the need to have a separate
+database/online service to store experiment metadata. If we run a few
+experiments:
+
+```bash
+$ dvc exp run --set-param train.learning_rate=0.1
+...
+$ dvc exp run --set-param train.learning_rate=0.01
+...
+$ dvc exp run --set-param train.batch_size=16
+...
+```
+
+We can display all of them with the `dvc exp show` command:
+
+```bash
+$ dvc exp show --only-changed --sort-by=dice_mean
+ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+  Experiment                Created        train.loss   valid.loss   foreground.acc   jaccard.coeff   dice.multi   dice_mean   acc_mean   train.learning_rate   train.batch_size   models
+ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+  workspace                 -                 0.10356     0.069076          0.90321         0.75906      0.92371     0.70612    0.97689   0.01                  16                 5854528
+  exp                       Apr 09, 2022      0.13305     0.087599          0.77803         0.66494      0.89084     0.70534    0.97891   0.01                  8                  6c513ae
+  ├── 83a4975 [exp-2d80e]   Apr 09, 2022      0.11189     0.088695          0.86905         0.75296      0.92005     0.70612    0.97689   0.01                  16                 5854528
+  ├── 675efb3 [exp-6c274]   Apr 09, 2022      0.10356     0.069076          0.90321         0.75906      0.92371     0.71492    0.98099   0.1                   16                 770745a
+  └── c8b1857 [exp-04bcd]   Apr 09, 2022      0.11189     0.088695          0.86905         0.75296      0.92005     0.71619    0.98025   0.01                  8                  094c420
+ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+```
+
+Then we’d pick the best one (e.g. highest `dice_mean` score) and commit it to
+our git repo:
```

Comment:
I think wording here might be misleading (commit has a meaning in git context that doesn't match what exp push do).

Not sure how to phrase it, but we could explicitly use the words persist and share, with links to respective docs pages (https://dvc.org/doc/user-guide/experiment-management/persisting-experiments).

Reply:
And perhaps only to clarify (as people might not be aware) also explicitly mention that there is no requirement for picking a single experiment, multiple experiments can be shared / persisted

Reply:
Good point! Updated.